File: tech-dhtml/revisarCodigo/game-connect4-main/src/types/Color.js
export class Color {

    static RED = new Color(`Red`);
    static YELLOW = new Color(`Yellow`);
    static NULL = new Color(` `);
    #string;

    constructor(string) {
        this.#string = string;
    }

    static values() {
        return [Color.RED, Color.YELLOW];
    }

    static get(ordinal) {
        return Color.values()[ordinal];
    }

    toString() {
        return this.#string;
    }

}

File: tech-java-orientadoObjetos/src/main/es/labturing/connect4/types/Color.java
package main.es.labturing.connect4.types;

public enum Color {

    RED,
    YELLOW,
    NULL;

    private Color() {

    }

    public static Color get(int ordinal) {
        return Color.values()[ordinal];
    }

    public Color getOpposite() {
        return Color.values()[this.ordinal() % (Color.values().length - 1)];
    }

}
